lib/activerecord-bulk_update/activerecord/relation.rb

```diff
@@ -4,18 +4,38 @@ module ActiveRecord
   class BulkInvalid < ActiveRecordError; end
 
   class Relation
+
+    # Inserts multiple validated records into the database in a single query.
+    #
+    # @example
+    #
+    # users = [User.new(name: "foo"), User.new(name: "bar")]
+    # User.where(active: true).bulk_create(users)
+    #
+    # @param [Record] inserts The records to insert.
+    # @param [Object] validate If truthy the records are validated.
+    # @param [Object] ignore_persisted If truthy any persisted records are ignored.
+    #
+    # @see {#bulk_insert}
+    #
+    # @raise [ActiveRecord::ActiveRecordError] if inserts contains persisted records and ignore_persisted is false.
+    # @raise [ActiveRecord::RecordNotUnique] if inserts contains duplicated records.
+    # @raise [ActiveRecord::BulkInvalid] if inserts contains invalid records and validate if truthy.
+    #
+    # @return [ActiveRecord::Result] containing the records as they have been inserted.
     def bulk_create(*args, **kwargs)
       bulk_create!(*args, **kwargs)
     rescue ActiveRecord::BulkInvalid
-      false
+      nil
```

Comment:
Ik twijfel nog wat over deze `nil`. Had je dit gedrag ook ergens van overgenomen?

Reply:
Dit is in lijn met het gedrag van `create` op gewone models:
```rb
relation = Relation.create!(name: 'foo', email_address: 'foo@mail.com', company: Company.take);
result = Relation.create(name: relation.name, email_address: relation.email_address, company: relation.company);
```

```
development> result.nil?
 => true
```

Reply:
Ik had dit zelf op de volgende manier getest en dan krijg je een unpersisted object terug:
```ruby
development> label = Label.create(administration_id: Administration.take.id)
  Administration Load (1.7ms)  SELECT "administrations".* FROM "administrations" WHERE "administrations"."marked_for_deletion_at" IS NULL LIMIT $1  [["LIMIT", 1]]
 => 
#<Label:0x0000000162dd75d0
...
development> label
 => 
#<Label:0x0000000162dd75d0
 id: nil,
 administration_id: 254,
 name: nil,
 credit_case_labels_count: 0,
 debtor_labels_count: 0,
 invoice_labels_count: 0,
 created_at: nil,
 updated_at: nil,
 color: "#e0e2e9",
 icon: nil,
 insured_labels_count: 0,
 label_standard_tasks_count: 0>
 ```

Reply:
Verschil zit hem waarschijnlijk in de waarom, jouw voorbeeld klapt op een active record validation, denk ik, terwijl dat van mij klapt op een database constraint. 

Gegeven dat we `ActiveRecord::BulkInsert` catchen, en die error raisen als de records niet valid zijn volgens de ActiveRecord validatie lijkt het me logisch om net als in `create` de unperisted records te returnen, dat is dan het meest in lijn met hoe ActiveRecord het doet. 

Eens?

Reply:
Yes eens!

Reply:
Staat klaar!